packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -888,7 +888,7 @@ export interface SharedObjectKind<out TSharedObject = unknown>
  * See {@link @fluidframework/fluid-static#ContainerSchema} for how this is used in the declarative API.
  * @internal
  */
-export function createSharedObjectKind<TSharedObject>(
+export function createSharedObjectKind<TSharedObject extends IFluidLoadable>(
 	factory: (new () => IChannelFactory<TSharedObject>) & { readonly Type: string },
```

Comment:
Adding `extends IFluidLoadable` to TSharedObject here might make more sense than forcing it into the TSharedObject which is typically part of the exported type, as this is an internal requirement.

As for fixing is, I think making a robust runtime way to make it work correctly with an input thats `unknown` would be ideal, and not too impractical.